pkg/apis/apps/v1alpha1/apicast_types.go

```diff
@@ -13,14 +15,95 @@ type APIcastSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Replicas                  *int64              `json:"replicas"`
+	AdminPortalCredentialsRef *v1.SecretReference `json:"adminPortalCredentialsRef"`
+	// +optional
+	EnvironmentConfigurationSecretRef *v1.SecretEnvSource `json:"environmentConfigurationSecretRef,omitempty"`
+	// +optional
+	ServiceAccount *string `json:"serviceAccount,omitempty"`
+	// +optional
+	Image *string `json:"image,omitempty"`
+	// +optional
+	ExposedHost *APIcastExposedHost `json:"exposedHost,omitempty"`
+
+	// +optional
+	DeploymentEnvironment *DeploymentEnvironmentType `json:"deploymentEnvironment,omitempty"` // THREESCALE_DEPLOYMENT_ENV
```

Comment:
I tried to define a type for the attributes that only accept a set of values but it seems that operator-sdk generator does not autogenerate openapi specs that limit to those value so I'm not really sure whether this is worth it, as it has some consequences (tying to a type, having to convert to string sometimes etc.)